include/SDL3/SDL_gpu.h

```diff
@@ -0,0 +1,2251 @@
+} SDL_GpuStencilOp;
+
+typedef enum SDL_GpuBlendOp
+{
+    SDL_GPU_BLENDOP_ADD,
+    SDL_GPU_BLENDOP_SUBTRACT,
+    SDL_GPU_BLENDOP_REVERSE_SUBTRACT,
+    SDL_GPU_BLENDOP_MIN,
+    SDL_GPU_BLENDOP_MAX
+} SDL_GpuBlendOp;
+
+typedef enum SDL_GpuBlendFactor
+{
+    SDL_GPU_BLENDFACTOR_ZERO,
+    SDL_GPU_BLENDFACTOR_ONE,
+    SDL_GPU_BLENDFACTOR_SRC_COLOR,
+    SDL_GPU_BLENDFACTOR_ONE_MINUS_SRC_COLOR,
+    SDL_GPU_BLENDFACTOR_DST_COLOR,
+    SDL_GPU_BLENDFACTOR_ONE_MINUS_DST_COLOR,
+    SDL_GPU_BLENDFACTOR_SRC_ALPHA,
+    SDL_GPU_BLENDFACTOR_ONE_MINUS_SRC_ALPHA,
+    SDL_GPU_BLENDFACTOR_DST_ALPHA,
+    SDL_GPU_BLENDFACTOR_ONE_MINUS_DST_ALPHA,
+    SDL_GPU_BLENDFACTOR_CONSTANT_COLOR,
+    SDL_GPU_BLENDFACTOR_ONE_MINUS_CONSTANT_COLOR,
+    SDL_GPU_BLENDFACTOR_SRC_ALPHA_SATURATE
+} SDL_GpuBlendFactor;
+
+typedef enum SDL_GpuColorComponentFlagBits
+{
+    SDL_GPU_COLORCOMPONENT_R_BIT = 0x00000001,
+    SDL_GPU_COLORCOMPONENT_G_BIT = 0x00000002,
+    SDL_GPU_COLORCOMPONENT_B_BIT = 0x00000004,
+    SDL_GPU_COLORCOMPONENT_A_BIT = 0x00000008
+} SDL_GpuColorComponentFlagBits;
+
+typedef Uint32 SDL_GpuColorComponentFlags;
+
+typedef enum SDL_GpuFilter
+{
+    SDL_GPU_FILTER_NEAREST,
+    SDL_GPU_FILTER_LINEAR
+} SDL_GpuFilter;
+
+typedef enum SDL_GpuSamplerMipmapMode
+{
+    SDL_GPU_SAMPLERMIPMAPMODE_NEAREST,
+    SDL_GPU_SAMPLERMIPMAPMODE_LINEAR
+} SDL_GpuSamplerMipmapMode;
+
+typedef enum SDL_GpuSamplerAddressMode
+{
+    SDL_GPU_SAMPLERADDRESSMODE_REPEAT,
+    SDL_GPU_SAMPLERADDRESSMODE_MIRRORED_REPEAT,
+    SDL_GPU_SAMPLERADDRESSMODE_CLAMP_TO_EDGE
+} SDL_GpuSamplerAddressMode;
+
+/*
+ * VSYNC:
+ *   Waits for vblank before presenting.
+ *   If there is a pending image to present, the new image is enqueued for presentation.
+ *   Disallows tearing at the cost of visual latency.
+ *   When using this present mode, AcquireSwapchainTexture will block if too many frames are in flight.
+ * IMMEDIATE:
+ *   Immediately presents.
+ *   Lowest latency option, but tearing may occur.
+ *   When using this mode, AcquireSwapchainTexture will return NULL if too many frames are in flight.
+ * MAILBOX:
+ *   Waits for vblank before presenting. No tearing is possible.
+ *   If there is a pending image to present, the pending image is replaced by the new image.
+ *   Similar to VSYNC, but with reduced visual latency.
+ *   When using this mode, AcquireSwapchainTexture will return NULL if too many frames are in flight.
+ */
+typedef enum SDL_GpuPresentMode
+{
+    SDL_GPU_PRESENTMODE_VSYNC,
+    SDL_GPU_PRESENTMODE_IMMEDIATE,
+    SDL_GPU_PRESENTMODE_MAILBOX
+} SDL_GpuPresentMode;
+
+/*
+ * SDR:
+ *   B8G8R8A8 or R8G8B8A8 swapchain. Pixel values are in nonlinear sRGB encoding. Blends raw pixel values.
+ * SDR_LINEAR:
+ *   B8G8R8A8_SRGB or R8G8B8A8_SRGB swapchain. Pixel values are in nonlinear sRGB encoding. Blends in linear space.
+ * HDR_EXTENDED_LINEAR:
+ *   R16G16B16A16_SFLOAT swapchain. Pixel values are in extended linear encoding. Blends in linear space.
```

Comment:
Probably makes sense to give more details here, that it's Windows scRGB, etc:

```
          The expected mapping is such that a color value of (1.0, 1.0, 1.0)
          corresponds to a luminance level of 80 nits on a standardized studio
          monitor.
          As the color value per channel goes beyond 1.0 and up to 125.0, the
          corresponding luminance levels also increase linearly to
          a maximum of 10000 nits.
```

Reply:
Note that in macOS, [1.0 corresponds to the brightness of an SDR pixel with value 1.0](https://developer.apple.com/documentation/metal/hdr_content/performing_your_own_tone_mapping), which is not guaranteed to be 80 nits (and varies with the user's brightness settings).  Unless the user has their brightness set extremely low, 1.0 will usually be much higher than 80 nits.